frontend/src/Card.tsx

```diff
@@ -48,19 +51,25 @@ const Card = ({ card, onInputClick }) => {
       </div>
     )
   } else {
+    const shape = shapeLookup.get(card.shape)
+
+    if (shape === undefined) {
+      throw new Error('Card shape is undefined.')
+    }
```

Comment:
Adding a guard here makes TS happy that at any point past here, we can guarantee that `shape` is not undefined.